internal/tasks/callback_manager_test.go

```diff
@@ -0,0 +1,335 @@
+package tasks
+
+import (
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/flightctl/flightctl/internal/util"
+	flightlog "github.com/flightctl/flightctl/pkg/log"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+type MockPublisher struct {
+	publishCallCount int
+}
+
+func (m *MockPublisher) Publish(payload []byte) error {
+	m.publishCallCount++
+	return nil
+}
+
+func (m *MockPublisher) Close() {
+}
+
+func (m *MockPublisher) ResetPublishCallCount() {
+	m.publishCallCount = 0
+}
+
+var (
+	mockPublisher = &MockPublisher{}
+	logger        = flightlog.InitLogs()
+)
+
+var _ = Describe("FleetUpdatedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher.ResetPublishCallCount()
+	})
+
+	When("both before and after are nil", func() {
+		It("does nothing", func() {
+			callbackManager := NewCallbackManager(mockPublisher, logger)
+			callbackManager.FleetUpdatedCallback(nil, nil)
+			Expect(mockPublisher.publishCallCount).To(Equal(0))
+		})
+	})
+
+	When("before is nil and after is not nil", func() {
+		It("submits FleetValidateTask and FleetSelectorMatchTask", func() {
+			after := createTestFleet("after", `json:"spec"`, "selector")
+			callbackManager := NewCallbackManager(mockPublisher, logger)
+			callbackManager.FleetUpdatedCallback(nil, after)
+			Expect(mockPublisher.publishCallCount).To(Equal(2))
```

Comment:
Is it worth tracking and checking which callbacks are called?

Reply:
+1

Reply:
Done